src/shared.ts

```diff
@@ -162,6 +171,18 @@ export const initStripe = (
     return null;
   }
 
+  const pk = args[0];
+  const isTestKey = pk.match(/^pk_test/);
+
+  // @ts-expect-error this is not publicly typed
+  const version = runtimeVersionToUrlVersion(maybeStripe.version);
+  const expectedVersion = RELEASE_TRAIN;
+  if (isTestKey && version !== expectedVersion) {
+    console.warn(
+      `Stripe.js@${version} was loaded on the page, but @stripe/stripe-js@${_VERSION} expected Stripe.js@${expectedVersion}. This may result in unexpected behavior.`
```

Comment:
```suggestion
      `Stripe.js@${version} was loaded on the page, but @stripe/stripe-js@${_VERSION} expected Stripe.js@${expectedVersion}. This may result in unexpected behavior. For more information see https://docs.stripe.com/sdks/stripejs-versioning`
```